knowledge/data-sources/website/colly.go

```diff
@@ -212,13 +204,25 @@ func scrape(ctx context.Context, logOut *logrus.Logger, output *MetadataOutput,
 				}
 				linkURL = parsedLink
 			}
-			e.Request.Visit(linkURL.String())
+
+			if err := e.Request.Visit(linkURL.String()); err == nil {
```

Comment:
Only marked url as visited if it has visited all the link under the url. Otherwise we could be missing url on the next restart.